src/execution/sql/ddl_executors.cpp

```diff
@@ -139,7 +140,8 @@ bool DDLExecutors::DropIndexExecutor(const common::ManagedPointer<planner::DropI
 
 bool DDLExecutors::CreateIndex(const common::ManagedPointer<catalog::CatalogAccessor> accessor,
                                const catalog::namespace_oid_t ns, const std::string &name,
-                               const catalog::table_oid_t table, const catalog::IndexSchema &input_schema) {
+                               const catalog::table_oid_t table, const catalog::IndexSchema &input_schema,
+                               const common::ManagedPointer<terrier::transaction::TransactionContext> populate_txn) {
   // Request permission from the Catalog to see if this a valid namespace and table name
   const auto index_oid = accessor->CreateIndex(ns, table, name, input_schema);
```

Comment:
How about txns that start after t1 (starting time of this populate_index txn)?
How will those txns be able to run? 
Are those txns being blocked here as well? If not, are they operating on another index?